File: src/main/java/com/alco/pubslist/controller/RollsController.java
package com.alco.pubslist.controller;

import com.alco.pubslist.entities.Roll;
import com.alco.pubslist.services.RollService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import static com.alco.pubslist.security.SecurityConstants.ROLLS_URL;

@RestController
public class RollsController {

	@Autowired
	private RollService service;

	@GetMapping(ROLLS_URL + "/daily")
	public Roll getDailyRoll() {

		return service.rollOfTheDay();
	}

	@GetMapping(ROLLS_URL + "/manual")
	public Roll getManualRoll() {

		return service.manualRoll();
	}
}

File: src/main/java/com/alco/pubslist/controller/RollsController.kt
package com.alco.pubslist.controller

import com.alco.pubslist.entities.Roll
import com.alco.pubslist.security.SecurityConstants
import com.alco.pubslist.services.RollService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class RollsController {
	@Autowired
	private val service: RollService? = null

	@get:GetMapping(SecurityConstants.ROLLS_URL + "/daily")
	val dailyRoll: Roll
		get() = service!!.rollOfTheDay()

	@get:GetMapping(SecurityConstants.ROLLS_URL + "/manual")
	val manualRoll: Roll
		get() = service!!.manualRoll()

	@GetMapping(SecurityConstants.ROLLS_URL)
	fun all(): Iterable<Roll> {
		return service!!.findAll()
	}
}
